### ISSUE TITLE
Add API to change validation controller's validate trigger

### ISSUE THREAD

**Issue by bfil** [aurelia/validation#419]
I would expect to be able to only display the errors after the initial form submission when using the `change` validation trigger, or alternatively setting the trigger to `manual` and after the first submission changing the validation trigger to `change`.

These options are currently not supported, the validation binding behaviour seems to store the initial trigger internally, so subsequent trigger updates don't change the behaviour, and there's no other way to prevent showing the errors until the initial form submission.

I'm not sure if there's another way to achieve this, but I've been playing around for quite a while now.

Just to clarify, the goal would be preventing displaying errors until I submit the form the first time but being able to keep using the `change` validation trigger behaviour after the first form submission.

To be honest, I found a way to do it, but it relies on a custom flag to track the initial submission and lots of boilerplate where the errors are displayed to prevent displaying them based on the flag, not ideal in my opinion.

**Comment by jdanyow** [comment]
Here's something you could do:

```js
  submit() {
    this.changeTrigger(validateTrigger.change);
    this.controller.validate();
  }
  
  changeTrigger(newTrigger) {
    this.controller.validateTrigger = newTrigger;
    const bindings = Array.from(this.controller.bindings.keys());
    for (let binding of bindings) {
      const source = binding.source;
      binding.unbind();
      binding.bind(source);
    }
  }
```

https://gist.run/?id=b795b2d5f22a8b6d11d155312dfcf009

**Comment by jdanyow** [comment]
This is a nice behavior by the way, thanks for submitting this issue. 👍

**Comment by bfil** [comment]
Nice! Thanks for the snippet @jdanyow 

I've actually tried to look at the source code and see if I could perform an unbind/bind on the ValidationController to reload the trigger but had no clue what else I could break by doing so. This looks helpful! 👍 

It could probably be part of the API of the ValidationController if it wouldn't cause any issues, it's a decent way to cover this scenario. For users with the `change` trigger scenario and for example an email validation it's quite annoying to see errors popping out as you type :) But it's nice to see the errors go away as soon as you fix the input value. So yeah, it could be a nice addition without heavy API changes.

I've actually wondered why there was no method to change the validation trigger in the first place and assigning a property on the controller was the way to do it, a `changeTrigger()` method would look much better, imo of course.

**Comment by jdanyow** [comment]
yep- will add this to the controller API. In the meantime, please let me know if you run into any issues with the workaround. Thanks!

**Comment by bfil** [comment]
No issues so far 👍

**Comment by jdanyow** [comment]
implemented in https://github.com/aurelia/validation/commit/d2aca549df187c8ac63c407f97d39828fea0f7c1

**Comment by robinchesterman** [comment]
With this it's now possible to change the trigger for the whole validation controller, but is it possible to change the trigger for individual fields? (It's possible to have initial triggers for each field set declaratively e.g. `& validateOnChange`.)

My preferred behaviour would be for each field to initially validate on blur (I really dislike validation errors like "please enter a valid email" when you've just started entering it) - but after a blur event and a field is invalid, it would then validate on change to show the user as soon as possible that the error has been corrected.

Basically a similar behaviour to what has been described in this issue, but instead of the trigger changing for the whole controller after the first form submission, each field can have its trigger changed when it has failed validation once and is now in a "correct me" state.

**Comment by bmcdaniels** [comment]
Stumbled across @robinchesterman's response as the only person bringing this up. I'm surprised it isn't complained about more often. If no support can be added to create this ourselves, at the very least can we get a new binding behavior, validateOnChangeAfterBlur?

**Comment by robinchesterman** [comment]
I know the Aurelia team are super busy - but is anyone looking at comments on issues? Even a simple reply to say “we’re not doing this” would be appreciated, otherwise it’s just left hanging in the ether.

**Comment by bigopon** [comment]
@robinchesterman I don't think anyone's comment is being ignored. It's just simply anticipated to be complex and take time to implement, while there is already workaround or enough lower level APIs to do it, it's low on priority. But your criticism holds, we should have better labeling activity to precisely describe priority.
@EisenbergEffect @jdanyow thoughts?

**Comment by EisenbergEffect** [comment]
Looks like this got missed with all the activity we have going on. @robinchesterman We don't have someone focusing on extending the validation library with new features right now. @jdanyow  was leading this project but he's had less time to devote in recent months. We'd be happy to take a PR to add the functionality that's requested. Since this is a bit different request than the original issue, would you mind opening a new issue to track the work?
